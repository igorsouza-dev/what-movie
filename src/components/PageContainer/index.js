import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
import HeaderBar from 'components/HeaderBar';
import Footer from 'components/Footer';
import {
  Container, Content, DrawerList, ListItem
} from './styles';

function PageContainer({ children }) {
  const showList = useSelector((state) => state.showList);
  const items = useSelector((state) => {
    if (showList === 'favorites') {
      return state.favorites;
    }
    return state.watchLater;
  });
  const itemsKeys = Object.keys(items);
  return (
    <Container>
      <HeaderBar />
      <DrawerList show={showList}>
        <h3>{showList === 'watchLater' ? 'Watch Later' : 'Favorites'}</h3>
        <ul>
          {itemsKeys.length
            ? itemsKeys.map((key) => (
              <Link to={`/details/${key}`} key={key}>
                <ListItem>{items[key].original_title}</ListItem>
              </Link>
            ))
            : 'No movies added.'}
        </ul>
      </DrawerList>
      <Content>{children}</Content>
      <Footer />
    </Container>
  );
}

PageContainer.propTypes = {
  children: PropTypes.element.isRequired,
};

export default PageContainer;
