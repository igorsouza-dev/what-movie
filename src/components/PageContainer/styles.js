import styled from 'styled-components';
import { screenSize } from 'styles/variables';

const drawerSize = 300;
export const Container = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-direction: column;
  position: relative;
`;

export const Content = styled.div`
  width: 100%;
  display: flex;
  padding-top: 80px;
  justify-content: flex-start;
  align-items: center;
  flex-direction: column;
  flex: 1;
  max-width: 900px;

  @media only screen and (max-width: ${screenSize.mobileS}) {
    max-width: 100%;
  }
  @media only screen and (min-width: 321px) and (max-width: ${screenSize.mobileM}) {
    max-width: ${screenSize.mobileM};
  }
  @media only screen and (min-width: 376px) and (max-width: ${screenSize.mobileL}) {
    max-width: ${screenSize.mobileL};
  }
  @media only screen and (min-width: 426px) and (max-width: 767px)  {
    max-width: 767px;
  }
  @media only screen and (min-width: 768px) and (max-width: ${screenSize.laptop})  {
    max-width: ${screenSize.laptop};
  }
  @media only screen and (min-width: 1025px) {
    max-width: ${screenSize.desktop};
  }
`;

export const DrawerList = styled.div`
  position: fixed;
  top: 55px;
  right: ${(props) => (props.show ? '0' : `-${drawerSize}px`)};
  height: 100%;
  z-index: 9;
  width: ${drawerSize}px;
  background: #fff;
  box-shadow: 0 0 25px 10px rgba(0, 0, 0, 0.3);
  padding: 10px;
`;

export const ListItem = styled.li`
  padding: 10px;
  width: 100%;
  &:hover {
    cursor: pointer;
    background: #ddd;
  }
`;
