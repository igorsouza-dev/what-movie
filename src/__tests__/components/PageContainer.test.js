import React from 'react';
import { cleanup } from '@testing-library/react';
import PageContainer from 'components/PageContainer';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import { renderWithRedux } from '../utils/redux';

describe('<PageContainer />', () => {
  afterEach(cleanup);
  it('should render', () => {
    const history = createMemoryHistory();
    const { container } = renderWithRedux(
      <Router history={history}>
        <PageContainer>
          <span>Some content</span>
        </PageContainer>
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render some content inside the container', () => {
    const history = createMemoryHistory();
    const { getByText } = renderWithRedux(
      <Router history={history}>
        <PageContainer>
          <span>Some content</span>
        </PageContainer>
      </Router>
    );
    expect(getByText(/some content/i)).toBeInTheDocument();
  });
  it('should show the movie list drawer with favorites', () => {
    const history = createMemoryHistory();
    const { getByText } = renderWithRedux(
      <Router history={history}>
        <PageContainer>
          <span>Testing side drawer</span>
        </PageContainer>
      </Router>,
      {
        initialState: {
          favorites: {
            1234: { id: 1234, original_title: 'Movie 1' },
            1235: { id: 1235, original_title: 'Movie 2' },
          },
          watchLater: {},
          showList: 'favorites',
        },
      }
    );
    expect(getByText(/Movie 1/i)).toBeInTheDocument();
    expect(getByText(/Movie 2/i)).toBeInTheDocument();
  });
  it('should show the movie list drawer with watch later', () => {
    const history = createMemoryHistory();
    const { getByText } = renderWithRedux(
      <Router history={history}>
        <PageContainer>
          <span>Testing side drawer</span>
        </PageContainer>
      </Router>,
      {
        initialState: {
          watchLater: {
            1234: { id: 1234, original_title: 'Movie 1' },
            1235: { id: 1235, original_title: 'Movie 2' },
          },
          favorites: {},
          showList: 'watchLater',
        },
      }
    );
    expect(getByText(/Movie 1/i)).toBeInTheDocument();
    expect(getByText(/Movie 2/i)).toBeInTheDocument();
  });
});
