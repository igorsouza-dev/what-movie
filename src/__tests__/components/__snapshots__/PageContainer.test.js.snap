// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageContainer /> should render 1`] = `
<div
  class="sc-iwsKbI gWyLDK"
>
  <div
    class="sc-htoDjs dJCVbN"
  >
    <a
      class="sc-dnqmqq lcgmO"
      href="/"
    >
      <h1
        class="sc-bxivhb gZfZeg"
        color="#fff"
        data-testid="logo"
        size="40"
      >
        What Movie?
      </h1>
    </a>
    <form
      class="sc-bdVaJa hjizjZ"
      data-testid="search-bar"
    >
      <input
        class="sc-bwzfXH aYadF"
        placeholder="Type the name of a movie"
        type="text"
        value=""
      />
      <button
        class="sc-htpNat jisJQc"
        type="submit"
      >
        <svg
          color="#fff"
          fill="currentColor"
          height="24"
          size="24"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(255, 255, 255);"
          viewBox="0 0 512 512"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
      </button>
    </form>
    <button
      class="sc-ifAKCX gTwPhi"
      data-testid="watch-later"
    >
      <svg
        color="#fff"
        fill="currentColor"
        height="24"
        size="24"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 512 512"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
        />
      </svg>
      <div
        class="sc-EHOje sBAZD"
        data-testid="count-value"
      >
        0
      </div>
    </button>
    <button
      class="sc-bZQynM kaqpnu"
      data-testid="favorites-btn"
    >
      <svg
        color="#fff"
        fill="currentColor"
        height="24"
        size="24"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 576 512"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
        />
      </svg>
      <div
        class="sc-gzVnrw ibxOwE"
        data-testid="count-value"
      >
        0
      </div>
    </button>
  </div>
  <div
    class="sc-gqjmRU jfNrNN"
  >
    <h3>
      Favorites
    </h3>
    <ul>
      No movies added.
    </ul>
  </div>
  <div
    class="sc-gZMcBi fAhRbv"
  >
    <span>
      Some content
    </span>
  </div>
  <div>
    Film data provided by TMDB
  </div>
</div>
`;
